src/main/java/net/contargo/osrmproxy/router/OsrmProfileRouter.java

```diff
@@ -1,137 +1,20 @@
-
-    @Autowired
-    public OsrmProfileRouter(ProfilesProperties profilesProperties,
-        AerialDistanceCalculationService aerialDistanceCalculationService) {
-
-        this.limitInMeters = profilesProperties.getLimitInMeters();
-        this.destinationOverLimit = profilesProperties.getDestinationOverLimit();
-        this.destinationUnderLimit = profilesProperties.getDestinationUnderLimit();
-        this.fallbackDestination = profilesProperties.getFallbackDestination();
-        this.aerialDistanceCalculationService = aerialDistanceCalculationService;
-    }
-
-    @Override
-    public String filterType() {
-
-        return ROUTE_TYPE;
-    }
-
-
-    @Override
-    public int filterOrder() {
-
-        return 0;
-    }
-
-
-    @Override
-    public boolean shouldFilter() {
-
-        return true;
-    }
-
-
-    @Override
-    public Object run() {
-
-        RequestContext ctx = RequestContext.getCurrentContext();
-
-        Boolean isRoutingRequest = ctx.getRequest().getRequestURI().startsWith(ROUTING_URL_PREFIX);
-        URL destination = isRoutingRequest ? getDestinationForRouting(ctx) : fallbackDestination;
-        ctx.setRouteHost(destination);
-
-        LOG.debug("Proxy destination {}", destination);
-
-        return null;
-    }
-
-
-    private URL getDestinationForRouting(RequestContext ctx) {
-
-        URL destination = null;
-
-        try {
-            double distance = calculateAerialDistanceFromUrl(ctx.getRequest().getRequestURI());
-            destination = distance > limitInMeters ? destinationOverLimit : destinationUnderLimit;
-        } catch (NoCoordinatesException e) {
-            LOG.error("No coordinates were found: {}", e.getRequestURI(), e);
-        } catch (IllegalStateException | IllegalArgumentException e) {
-            LOG.error("Calculation of aerial distance {} had errors", ctx.getRequest().getRequestURI(), e);
-        }
-
-        // In case of an error (e.g. NoCoordinatesException, IllegalStateException or IllegalArgumentException )
-        // the request should be proxied to the default destination
-        return destination == null ? fallbackDestination : destination;
-    }
-
-
-    private double calculateAerialDistanceFromUrl(String requestURI) {
-
-        Matcher matcher = COORDINATE_PATTERN.matcher(requestURI);
-
-        if (matcher.matches()) {
-            double lonA = parseDouble(matcher.group(INDEX_LON_A));
-            double latA = parseDouble(matcher.group(INDEX_LAT_A));
-            double lonB = parseDouble(matcher.group(INDEX_LON_B));
-            double latB = parseDouble(matcher.group(INDEX_LAT_B));
-
-            return aerialDistanceCalculationService.calculateAerialDistanceInMeters(lonA, latA, lonB, latB);
-        } else {
-            throw new NoCoordinatesException(requestURI);
-        }
-    }
+public interface OsrmProfileRouter {
+
+    /**
+     * Returns the host with port and protocol, to which the request needs to be redirected.
```

Comment:
`Returns the host with port and protocol to which requests should be redirected.`?